internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -748,7 +761,16 @@ func setRootFields(d *schema.ResourceData, cluster *admin.ClusterDescription2024
 	if err := d.Set("global_cluster_self_managed_sharding", cluster.GetGlobalClusterSelfManagedSharding()); err != nil {
 		return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "global_cluster_self_managed_sharding", clusterName, err))
 	}
-
+	if cluster.ConfigServerType != nil {
+		if err := d.Set("config_server_type", *cluster.ConfigServerType); err != nil {
+			return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "config_server_type", clusterName, err))
+		}
+	}
+	if cluster.ConfigServerManagementMode != nil {
+		if err := d.Set("config_server_management_mode", *cluster.ConfigServerManagementMode); err != nil {
```

Comment:
can we simplify this by using SDK getter functions? Same for above

Reply:
You were right. Changed in c38307cf8c1661f9bb0160a9d724263fc5c87473
I thought we didn't want to update state when the user hasn't set those fields before